src/AutoRest.CSharp/MgmtTest/TestCommon/Tool.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text;
+using Newtonsoft.Json;
+using System.Text.Json;
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    internal static class Tool
+    {
+        public static FormattableString? RefScenariDefinedVariables(this string? src, IEnumerable<string>? scenarioVariables)
+        {
+            if (src is null)
+            {
+                return null;
+            }
+            if (scenarioVariables is null)
+            {
+                return $"{src:L}";
+            }
+
+            StringBuilder stringBuilder = new StringBuilder(src);
+            stringBuilder = stringBuilder.Replace(@"{", @"{{").Replace(@"}", @"}}");
+            foreach (var scenarioVariable in scenarioVariables)
+            {
+                stringBuilder.Replace($"$({scenarioVariable})", $"{{{scenarioVariable}}}");
+            }
+            return $"${stringBuilder.ToString():L}";
+        }
+
+        public static string ToJsonString(this JsonDocument jdoc)
+        {
+            using (var stream = new MemoryStream())
+            {
+                Utf8JsonWriter writer = new Utf8JsonWriter(stream, new JsonWriterOptions { Indented = true });
+                jdoc.WriteTo(writer);
+                writer.Flush();
+                return Encoding.UTF8.GetString(stream.ToArray());
+            }
+        }
+
+        public static string ToSnakeCase(this string text)
```

Comment:
We might have something similar in `Common/Utilities/StringExtentions.cs`, could you please find if there is a function doing the same thing?

Reply:
checked SplitByCamelCase or ToCleanName don't serve for the same purpose with 'snake case'.